### PR TITLE
Remove empty path segments when joining paths

### PR DESCRIPTION
**Pull request by jhendrixMSFT**
- [ ] The purpose of this PR is explained in this or a referenced issue.
- [ ] The PR does not update generated files.
   - These files are managed by the codegen framework at [Azure/autorest.go][].
- [ ] Tests are included and/or updated for code changes.
- [ ] Updates to [CHANGELOG.md][] are included.
- [ ] MIT license headers are included in each file.

[Azure/autorest.go]: https://github.com/Azure/autorest.go
[CHANGELOG.md]: https://github.com/Azure/azure-sdk-for-go/blob/main/CHANGELOG.md

### CODE REVIEW

File: sdk/azcore/runtime/request.go [hunk 1]
```diff
@@ -59,6 +59,10 @@ func JoinPaths(root string, paths ...string) string {
 	for i := 0; i < len(paths); i++ {
 		root = strings.TrimRight(root, "/")
 		paths[i] = strings.TrimLeft(paths[i], "/")
+		// remove any empty path segments. this can happen when an
+		// optional path param is unspecified so its placeholder is
+		// replaced with the empty string.
+		paths[i] = strings.Replace(paths[i], "//", "/", -1)
```
[the remark this answers is not in the document]
Comment (richardpark-msft) [reply sdk/azcore/runtime/request.go:65]: Also, if paths[i] == "" we should skip it, right?

**Comment by chlowell** [comment]
I noticed this while working with Key Vault clients but let it lie because empty path segments are legal. Does this change fix a particular scenario?

File: sdk/azcore/runtime/request.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (jhendrixMSFT) [reply sdk/azcore/runtime/request.go:65]: The typical case where we see this is with optional path params that are empty.  Here's an example.

[It's not that they were prejoined (although that could](https://github.com/Azure/azure-sdk-for-go/blob/main/sdk/resourcemanager/resources/armresources/zz_generated_client.go#L193)

Here, `parentResourcePath` is allowed to be the empty string.  So when it is, the `{parentResourcePath}` token is replaced with an empty string resulting in the empty path segment.

**Comment by jhendrixMSFT** [comment]
No particular scenario is broken, it's just something that's bugged me for a while and might end up being exposed to customers when faking (depending on what direction we go).

I didn't realize that empty path segments are legal.  Is there a reference for that in the RFC?  If so, then leaving them is actually advantageous for faking.

**Comment by jhendrixMSFT** [comment]
Ah found it.  This is great and fixes part of problem with fakes.

https://stackoverflow.com/questions/20569958/url-format-empty-path

https://www.rfc-editor.org/rfc/rfc3986#section-3.3

**Pull request closed without merging**